### PR TITLE
Add sticky header support to aside section

### PR DESCRIPTION
**Pull request by lukas-frey**
- [x] Changes have been thoroughly tested to not break existing functionality.
- [ ] New functionality has been documented or existing documentation has been updated to reflect changes.
- [x] Visual changes are explained in the PR description using a screenshot/recording of before and after.

This adds the possibility to make the description of an aside section sticky. If aside or sticky is set to false, it will behave like it did until now. Only if aside AND sticky are both true, it will stick to the top.

https://github.com/filamentphp/filament/assets/10926334/26786dd3-0b62-4bac-a4c8-f78ed60375f2

### CODE REVIEW

File: packages/forms/src/Components/Section.php [hunk 1]
```diff
@@ -21,6 +22,7 @@ class Section extends Component implements Contracts\CanConcealComponents, Contr
     use HasHeading;
     use HasIcon;
     use HasIconColor;
+    use CanBeStickied;
```
Comment (zepfietje) [inline packages/forms/src/Components/Section.php]: Should be `CanBeSticky`.

**Review by zepfietje — CHANGES_REQUESTED** [review]
It's really only the header that's sticky, right?
I think that should be reflected in the API.
With the current `sticky()` API I would expect sections to actually be sticky as a whole.

**Comment by lukas-frey** [comment]
> It's really only the header that's sticky, right? I think that should be reflected in the API. With the current `sticky()` API I would expect sections to actually be sticky as a whole.

Yes, only the header. So would you suggest something like `stickyHeader()` or `withStickyHeader()` instead?

**Comment by zepfietje** [comment]
`stickyHeader()` should work fine I think. Have a look at what's the exact API for sticky modal footers; we could use similar naming.

**Comment by lukas-frey** [comment]
Ready for review. The only thing I'm unsure about is whether the `CanBeSticky` name for the trait is still valid.

Since it defines whether the header is sticky, maybe something like `HasStickyHeader` or `CanHaveStickyHeader` or something is suited better?

**Comment by zepfietje** [comment]
I think `CanHaveStickyHeader` should be good. Though let's see what @danharrin thinks.

**Comment by zepfietje** [comment]
Also I'm thinking that technically you'd expect the header to be sticky even on mobile or when not `aside`.

**Comment by lukas-frey** [comment]
> Also I'm thinking that technically you'd expect the header to be sticky even on mobile or when not `aside`.

Yeah, but design-wise it doesn't make much sense imo, when there's a long description.

**Comment by danharrin** [comment]
Hi, thanks for the PR. I think it's important to give you a bit of information as to why this PR has been tagged as "low priority". We really appreciate all contributions, and intend to visit this in the future, and hopefully merge it if it is a suitable feature for the framework.

However, reviewing PRs, and often rewriting them on behalf of the author to ensure they are consistent with the rest of the framework, consumes a lot of my time. At the moment we have a large backlog of bug reports that I am prioritising, since most of these affect functionality that worked in a previous version and are now impacting real projects. Hopefully you can see how that work comes first before introducing new features that people aren't using in projects yet.

Again, thank you for contributing and I'm looking forward to reviewing this PR when our backlog of bug reports has reduced.

**Comment by zepfietje** [comment]
@danharrin, I think not just the `aside` section should support this. Let's try adding it to the regular section as well.

**Comment by danharrin** [comment]
I'm honestly not sure we should support this. It relies on the fact that the section headers are sticky `-20` from the top of the screen, which only really works in the admin panel where the topbar is that height. In another app, it might be a different distance.

I think this feature could be implemented quite easily in CSS within any app:

```php
Section::make('Heading')->extraAttributes(['class' => 'fi-section-sticky-header'])
```

```css
.fi-section-sticky-header .fi-section {
    @apply md:relative;
}

.fi-section-sticky-header .fi-section-header {
    @apply md:sticky md:start-0 md:top-20;
}
```

**Pull request closed without merging**